common/src/main/java/me/lucko/luckperms/common/locale/TranslationManager.java

```diff
@@ -132,6 +132,15 @@ private void loadFromResourceBundle() {
         }
     }
 
+    private ResourceBundle getResourceBundle() {
+        try {
+            return ResourceBundle.getBundle("luckperms", DEFAULT_LOCALE, UTF8ResourceBundleControl.get());
+        } catch (UnsupportedOperationException ex) {
+            // ResourceBundle.Control not supported in named modules
```

Comment:
What's the deal here? At the very least could we add a comment referencing the problem so regressions don't happen in future.

The UTF8ResourceBundleControl is necessary, as suggested, to make sure that the file is loaded with the correct encoding. (this wasn't the case on early Java versions)

We could switch the whole method to load the messages file as an InputStream/Properties object instead if that's more compatible.

Reply:
I believe that [modlauncher](https://github.com/McModLauncher/modlauncher) loads everything as modules which leads to `ResourceBundle.Control` throwing an `UnsupportedOperationException` as it doesn't support being called from a named module, I'm unsure as to why the `ResourceBundle.Control` has this limitation, however it should only affect Java 9+

https://docs.oracle.com/javase/9/docs/api/java/util/ResourceBundle.Control.html#note
"ResourceBundle.Control is not supported in named modules. If the ResourceBundle.getBundle method with a ResourceBundle.Control is called in a named module, the method will throw an UnsupportedOperationException."

> We could switch the whole method to load the messages file as an InputStream/Properties object instead if that's more compatible.

This may be the best way forwards as other platforms may switch to using the Java Module System and may run into the same issue.

Reply:
With the jar-in-jar system this is no longer an issue as the `TranslationManager` is no longer being loaded from a named module.

However it should be noted that this issue could reappear in the future as platforms advance to Java 9+ and adopt the Java Module System.